libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -226,6 +226,10 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 				"\"."
 			);
 	}
+
+	if (v050)
+		const_cast<FunctionDefinition &>(_function).optionallyMarkAsOverride();
```

Comment:
This is super ugly, but I couldn't come up with a better approach. We could clean this up for 0.5.0.